src/Airship/Resource.hs

```diff
@@ -139,3 +140,40 @@ defaultResource = Resource { allowMissingPost       = return False
                            , uriTooLong             = return False
                            , validContentHeaders    = return True
                            }
+
+setResourceParams :: Monad m => p -> Resource p m -> Resource p' m
+setResourceParams p r =
```

Comment:
FWIW This is pretty horrible. I don't see how to avoid this unless we, hypothetically, had a different representation of `Resource`